hazelcast-spring/src/main/resources/hazelcast-spring-3.11.xsd

```diff
@@ -2356,6 +2410,45 @@
         <xs:attribute name="discovery-strategy-factory" type="non-space-string" use="optional"/>
     </xs:complexType>
 
+    <xs:complexType name="wan-sync">
+        <xs:all>
+            <xs:element name="use-merkle-trees" type="parameterized-boolean" minOccurs="0" default="false">
+                <xs:annotation>
+                    <xs:documentation>
+                        Boolean denoting if merkle trees should be used to optimise WAN sync.
+                        The target cluster should support merkle trees for the sync procedure to
+                        function. That means:
+                        - the data structures being synced should be configured with merkle
+                        trees enabled both on the source and target cluster
+                        - a specific data structure (e.g. a single map) should have the same
```

Comment:
Actually, the trees can be set to different depths. At least, the nodes on the levels that both tree has will hold the same hashes with the same dataset. See https://github.com/hazelcast/hazelcast/pull/13450/files#diff-de06c875f5b227460b4d634c6fdf616eR358 and https://github.com/hazelcast/hazelcast/pull/13450/files#diff-de06c875f5b227460b4d634c6fdf616eR381
The linked tests verify if trees with different depths show difference only for the leaf that is intentionally made different in the two trees.